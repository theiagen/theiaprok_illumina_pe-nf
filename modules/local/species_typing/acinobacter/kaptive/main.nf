process KAPTIVE {
    tag "$meta.id"
    label "process_medium"

    container "us-docker.pkg.dev/general-theiagen/staphb/kaptive:2.0.3"

    input:
    tuple val(meta), path(assembly)
    val start_end_margin
    val min_percent_coverage
    val min_percent_identity
    val low_gene_percent_identity

    output:
    tuple val(meta), path("*_kaptive_out_k_table.tsv"), emit: k_table
    tuple val(meta), path("*_kaptive_out_oc_table.tsv"), emit: oc_table
    tuple val(meta), path("BEST_MATCH_LOCUS_K"), emit: k_match
    tuple val(meta), path("BEST_MATCH_TYPE_K"), emit: k_type
    tuple val(meta), path("MATCH_CONFIDENCE_K"), emit: k_confidence
    tuple val(meta), path("NUM_EXPECTED_INSIDE_K"), emit: k_expected_inside_count
    tuple val(meta), path("EXPECTED_GENES_IN_LOCUS_K"), emit: k_expected_inside_genes
    tuple val(meta), path("NUM_EXPECTED_OUTSIDE_K"), emit: k_expected_outside_count
    tuple val(meta), path("EXPECTED_GENES_OUT_LOCUS_K"), emit: k_expected_outside_genes
    tuple val(meta), path("NUM_OTHER_INSIDE_K"), emit: k_other_inside_count
    tuple val(meta), path("OTHER_GENES_IN_LOCUS_K"), emit: k_other_inside_genes
    tuple val(meta), path("NUM_OTHER_OUTSIDE_K"), emit: k_other_outside_count
    tuple val(meta), path("OTHER_GENES_OUT_LOCUS_K"), emit: k_other_outside_genes
    tuple val(meta), path("BEST_MATCH_LOCUS_OC"), emit: oc_match
    tuple val(meta), path("BEST_MATCH_TYPE_OC"), emit: oc_type
    tuple val(meta), path("MATCH_CONFIDENCE_OC"), emit: oc_confidence
    tuple val(meta), path("NUM_EXPECTED_INSIDE_OC"), emit: oc_expected_inside_count
    tuple val(meta), path("EXPECTED_GENES_IN_LOCUS_OC"), emit: oc_expected_inside_genes
    tuple val(meta), path("NUM_EXPECTED_OUTSIDE_OC"), emit: oc_expected_outside_count
    tuple val(meta), path("EXPECTED_GENES_OUT_LOCUS_OC"), emit: oc_expected_outside_genes
    tuple val(meta), path("NUM_OTHER_INSIDE_OC"), emit: oc_other_inside_count
    tuple val(meta), path("OTHER_GENES_IN_LOCUS_OC"), emit: oc_other_inside_genes
    tuple val(meta), path("NUM_OTHER_OUTSIDE_OC"), emit: oc_other_outside_count
    tuple val(meta), path("OTHER_GENES_OUT_LOCUS_OC"), emit: oc_other_outside_genes
    path "versions.yml", emit: versions

    when:
    task.ext.prefix || task.ext.stub || task.ext.skip

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def start_end_margin_arg = start_end_margin ? start_end_margin : 10
    def min_percent_coverage_arg = min_percent_coverage ? min_percent_coverage : 90.0
    def min_percent_identity_arg = min_percent_identity ? min_percent_identity : 80.0
    def low_gene_percent_identity_arg = low_gene_percent_identity ? low_gene_percent_identity : 95.0
    """
    KAPTIVE_DIR=\$(dirname "\$(which kaptive.py)")

    # Run for k locus
    kaptive.py \
    -t ${task.cpus} \
    --start_end_margin ${start_end_margin_arg} \
    --min_gene_id ${min_percent_identity_arg} \
    --min_gene_cov ${min_percent_coverage_arg} \
    --low_gene_id  ${low_gene_percent_identity_arg} \
    --no_seq_out \
    --no_json \
    --out ${prefix}_kaptive_out_k \
    --assembly ${assembly} \
    --k_refs \${KAPTIVE_DIR}/reference_database/Acinetobacter_baumannii_k_locus_primary_reference.gbk

    # Extract kaptive output tables
    kaptive_parser.py \
    --mode k \
    --input ${prefix}_kaptive_out_k_table.txt

    # Run for oc locus
    kaptive.py \
    -t ${task.cpus} \
    --start_end_margin ${start_end_margin_arg} \
    --min_gene_id ${min_percent_identity_arg} \
    --min_gene_cov ${min_percent_coverage_arg} \
    --low_gene_id  ${low_gene_percent_identity_arg} \
    --no_seq_out \
    --no_json \
    --out ${prefix}_kaptive_out_oc \
    --assembly ${assembly} \
    --k_refs \${KAPTIVE_DIR}/reference_database/Acinetobacter_baumannii_OC_locus_primary_reference.gbk

    # Extract kaptive output tables for oc locus
    kaptive_parser.py \
    --mode oc \
    --input ${prefix}_kaptive_out_oc_table.txt

    mv -v ${prefix}_kaptive_out_k_table.txt ${prefix}_kaptive_out_k_table.tsv
    mv -v ${prefix}_kaptive_out_oc_table.txt ${prefix}_kaptive_out_oc_table.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        kaptive: \$(kaptive.py --version)
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_kaptive_out_k_table.tsv
    touch ${prefix}_kaptive_out_oc_table.tsv
    touch BEST_MATCH_LOCUS_K
    touch BEST_MATCH_TYPE_K
    touch MATCH_CONFIDENCE_K
    touch NUM_EXPECTED_INSIDE_K
    touch EXPECTED_GENES_IN_LOCUS_K
    touch NUM_EXPECTED_OUTSIDE_K
    touch EXPECTED_GENES_OUT_LOCUS_K
    touch NUM_OTHER_INSIDE_K
    touch OTHER_GENES_IN_LOCUS_K
    touch NUM_OTHER_OUTSIDE_K
    touch OTHER_GENES_OUT_LOCUS_K
    touch BEST_MATCH_LOCUS_OC
    touch BEST_MATCH_TYPE_OC
    touch MATCH_CONFIDENCE_OC
    touch NUM_EXPECTED_INSIDE_OC
    touch EXPECTED_GENES_IN_LOCUS_OC
    touch NUM_EXPECTED_OUTSIDE_OC
    touch EXPECTED_GENES_OUT_LOCUS_OC
    touch NUM_OTHER_INSIDE_OC
    touch OTHER_GENES_IN_LOCUS_OC
    touch NUM_OTHER_OUTSIDE_OC
    touch OTHER_GENES_OUT_LOCUS_OC

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        kaptive: \$(kaptive.py --version)
    END_VERSIONS
    """
}