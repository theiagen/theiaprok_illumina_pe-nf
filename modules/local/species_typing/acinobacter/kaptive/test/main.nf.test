nextflow_process {
    name "Test Process KAPTIVE"
    script "../main.nf"
    process "KAPTIVE"

    // Grabbing staph b test asm
    test("kaptive - prokaryote assembly") {
        setup {
            run("GUNZIP") {
                script "../../../../../nf-core/gunzip/main.nf"
                process {
                    """
                    input[0] = [
                        [id:'test_sample'],
                        [file('https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/022/268/055/GCA_022268055.1_PDT000434809.1/GCA_022268055.1_PDT000434809.1_genomic.fna.gz', checkIfExists: true)]
                    ]
                    """
                }
            }
        }
        when {
            process {
                """
                input[0] = GUNZIP.out.gunzip
                input[1] = 10
                input[2] = 90
                input[3] = 80
                input[4] = 95
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.k_table, 
                    process.out.oc_table, 
                    process.out.k_match, 
                    process.out.k_type, 
                    process.out.k_confidence, 
                    process.out.k_expected_inside_count, 
                    process.out.k_expected_inside_genes, 
                    process.out.k_expected_outside_count, 
                    process.out.k_expected_outside_genes, 
                    process.out.k_other_inside_count, 
                    process.out.k_other_inside_genes, 
                    process.out.k_other_outside_count, 
                    process.out.k_other_outside_genes, 
                    process.out.oc_match, 
                    process.out.oc_type, 
                    process.out.oc_confidence, 
                    process.out.oc_expected_inside_count, 
                    process.out.oc_expected_inside_genes, 
                    process.out.oc_expected_outside_count, 
                    process.out.oc_expected_outside_genes, 
                    process.out.oc_other_inside_count, 
                    process.out.oc_other_inside_genes, 
                    process.out.oc_other_outside_count, 
                    process.out.oc_other_outside_genes,
                    process.out.versions
                ).match() }
            )
        }
    }

    test("kaptive - prokaryote assembly - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test_sample' ],
                    file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/genome/genome.fna.gz', checkIfExists: true)
                ])
                input[1] = 10
                input[2] = 90
                input[3] = 80
                input[4] = 95
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}