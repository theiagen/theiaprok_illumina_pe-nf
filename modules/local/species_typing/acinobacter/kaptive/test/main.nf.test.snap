{
    "kaptive - prokaryote assembly": {
        "content": [
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "test_sample_kaptive_out_k_table.tsv:md5,67c0f57b4ccb182dc4fc4b8f642779bc"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "test_sample_kaptive_out_oc_table.tsv:md5,abe925b36b036961517c975038621348"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "BEST_MATCH_LOCUS_K:md5,82b4557e30a61616f0d0ec14da312fab"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "BEST_MATCH_TYPE_K:md5,75fd026b16b90ad5c056cf31cba5af75"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "MATCH_CONFIDENCE_K:md5,6adf97f83acf6453d4a6a4b1070f3754"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "NUM_EXPECTED_INSIDE_K:md5,103cb539f94c65b4546f791171a46b1e"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "EXPECTED_GENES_IN_LOCUS_K:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "NUM_EXPECTED_OUTSIDE_K:md5,103cb539f94c65b4546f791171a46b1e"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "EXPECTED_GENES_OUT_LOCUS_K:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "NUM_OTHER_INSIDE_K:md5,cfcd208495d565ef66e7dff9f98764da"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "OTHER_GENES_IN_LOCUS_K:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "NUM_OTHER_OUTSIDE_K:md5,cfcd208495d565ef66e7dff9f98764da"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "OTHER_GENES_OUT_LOCUS_K:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "BEST_MATCH_LOCUS_OC:md5,701b720e336886983d2e553bd9eb5ff4"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "BEST_MATCH_TYPE_OC:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "MATCH_CONFIDENCE_OC:md5,6adf97f83acf6453d4a6a4b1070f3754"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "NUM_EXPECTED_INSIDE_OC:md5,73a4238f772a4ce623d365c59d457d7a"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "EXPECTED_GENES_IN_LOCUS_OC:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "NUM_EXPECTED_OUTSIDE_OC:md5,73a4238f772a4ce623d365c59d457d7a"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "EXPECTED_GENES_OUT_LOCUS_OC:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "NUM_OTHER_INSIDE_OC:md5,cfcd208495d565ef66e7dff9f98764da"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "OTHER_GENES_IN_LOCUS_OC:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "NUM_OTHER_OUTSIDE_OC:md5,cfcd208495d565ef66e7dff9f98764da"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "OTHER_GENES_OUT_LOCUS_OC:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-06-24T18:25:15.496273648"
    },
    "kaptive - prokaryote assembly - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample_kaptive_out_k_table.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample_kaptive_out_oc_table.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "10": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "OTHER_GENES_IN_LOCUS_K:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "11": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "NUM_OTHER_OUTSIDE_K:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "12": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "OTHER_GENES_OUT_LOCUS_K:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "13": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "BEST_MATCH_LOCUS_OC:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "14": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "BEST_MATCH_TYPE_OC:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "15": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "MATCH_CONFIDENCE_OC:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "16": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "NUM_EXPECTED_INSIDE_OC:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "17": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "EXPECTED_GENES_IN_LOCUS_OC:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "18": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "NUM_EXPECTED_OUTSIDE_OC:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "19": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "EXPECTED_GENES_OUT_LOCUS_OC:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "BEST_MATCH_LOCUS_K:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "20": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "NUM_OTHER_INSIDE_OC:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "21": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "OTHER_GENES_IN_LOCUS_OC:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "22": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "NUM_OTHER_OUTSIDE_OC:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "23": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "OTHER_GENES_OUT_LOCUS_OC:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "BEST_MATCH_TYPE_K:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "MATCH_CONFIDENCE_K:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "NUM_EXPECTED_INSIDE_K:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "EXPECTED_GENES_IN_LOCUS_K:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "7": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "NUM_EXPECTED_OUTSIDE_K:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "8": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "EXPECTED_GENES_OUT_LOCUS_K:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "9": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "NUM_OTHER_INSIDE_K:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "k_confidence": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "MATCH_CONFIDENCE_K:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "k_expected_inside_count": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "NUM_EXPECTED_INSIDE_K:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "k_expected_inside_genes": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "EXPECTED_GENES_IN_LOCUS_K:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "k_expected_outside_count": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "NUM_EXPECTED_OUTSIDE_K:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "k_expected_outside_genes": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "EXPECTED_GENES_OUT_LOCUS_K:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "k_match": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "BEST_MATCH_LOCUS_K:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "k_other_inside_count": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "NUM_OTHER_INSIDE_K:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "k_other_inside_genes": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "OTHER_GENES_IN_LOCUS_K:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "k_other_outside_count": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "NUM_OTHER_OUTSIDE_K:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "k_other_outside_genes": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "OTHER_GENES_OUT_LOCUS_K:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "k_table": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample_kaptive_out_k_table.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "k_type": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "BEST_MATCH_TYPE_K:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "oc_confidence": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "MATCH_CONFIDENCE_OC:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "oc_expected_inside_count": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "NUM_EXPECTED_INSIDE_OC:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "oc_expected_inside_genes": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "EXPECTED_GENES_IN_LOCUS_OC:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "oc_expected_outside_count": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "NUM_EXPECTED_OUTSIDE_OC:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "oc_expected_outside_genes": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "EXPECTED_GENES_OUT_LOCUS_OC:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "oc_match": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "BEST_MATCH_LOCUS_OC:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "oc_other_inside_count": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "NUM_OTHER_INSIDE_OC:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "oc_other_inside_genes": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "OTHER_GENES_IN_LOCUS_OC:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "oc_other_outside_count": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "NUM_OTHER_OUTSIDE_OC:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "oc_other_outside_genes": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "OTHER_GENES_OUT_LOCUS_OC:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "oc_table": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample_kaptive_out_oc_table.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "oc_type": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "BEST_MATCH_TYPE_OC:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-06-24T18:25:28.501748995"
    }
}